packages/peregrine/lib/talons/CheckoutPage/useCheckoutPage.js

```diff
@@ -54,25 +55,41 @@ export const useCheckoutPage = props => {
         fetchPolicy: 'network-only'
     });
 
-    const [
-        getCheckoutDetails,
-        {
-            data: checkoutData,
-            called: checkoutCalled,
-            client,
-            loading: checkoutLoading
+    const {
+        data: checkoutData,
+        networkStatus: checkoutQueryNetworkStatus
+    } = useQuery(getCheckoutDetailsQuery, {
+        /**
+         * Skip fetching checkout details if the `cartId`
+         * is a falsy value.
+         */
+        skip: !cartId,
+        notifyOnNetworkStatusChange: true,
+        variables: {
+            cartId
         }
-    ] = useLazyQuery(getCheckoutDetailsQuery);
+    });
+
+    /**
+     * For more info about network statues check this out
+     *
+     * https://www.apollographql.com/docs/react/data/queries/#inspecting-loading-states
+     */
+    const isLoading = useMemo(
+        () =>
+            checkoutQueryNetworkStatus ? checkoutQueryNetworkStatus < 7 : true,
```

Comment:
The comment is nice but I'd also like us to have a utility enumeration that exports Apollo's network status codes to make this more readable.

https://github.com/apollographql/apollo-client/blob/master/src/core/networkStatus.ts

This would become something like

```js
checkoutQueryNetworkStatus ? checkoutQueryNetworkStatus < ApolloNetworkStatuses.READY : true
```

Reply:
That source file also exposes an `isNetworkStatusInFlight` utility function that seems to be performing the same check you are here. Other places in our code could benefit from this as well.

Reply:
Nice idea @supernova-at. Will do.

Also minor adjustments to the comment above, I don't think we should export the network status from this talon. It can be something from a util or defaults file. I can do that if that is what you are expecting here.

Also another point, `isNetworkStatusInFlight` differs from our implementation. The else case in `isNetworkStatusInFlight` returns false but in our case it has to return true.

Reply:
Yeah, I was definitely thinking a separate file that this and other talons could use.